espnet/bin/tts_train.py

```diff
@@ -62,13 +63,23 @@ def main(cmd_args):
                         help="How many epochs to use sortagrad for. 0 = deactivated, -1 = all epochs")
     parser.add_argument('--batch_sort_key', default='shuffle', type=str,
                         choices=['shuffle', 'output', 'input'], nargs='?',
-                        help='Batch sorting key')
-    parser.add_argument('--batch-size', '-b', default=32, type=int,
-                        help='Batch size')
-    parser.add_argument('--maxlen-in', default=100, type=int, metavar='ML',
-                        help='Batch size is reduced if the input sequence length > ML')
-    parser.add_argument('--maxlen-out', default=200, type=int, metavar='ML',
-                        help='Batch size is reduced if the output sequence length > ML')
+                        help='Batch sorting key. "shuffle" only work with --batch-count "seq".')
+    parser.add_argument('--batch-count', default='auto', choices=BATCH_COUNT_CHOICES,
+                        help='How to count batch_size. The default (auto) will find how to count by args.')
+    parser.add_argument('--batch-size', '-b', default=0, type=int,
+                        help='Maximum seqs in a minibatch (0 to disable)')
+    parser.add_argument('--batch-bins', default=0, type=int,
+                        help='Maximum bins in a minibatch (0 to disable)')
+    parser.add_argument('--batch-frames-in', default=0, type=int,
+                        help='Maximum input frames in a minibatch (0 to disable)')
+    parser.add_argument('--batch-frames-out', default=0, type=int,
+                        help='Maximum output frames in a minibatch (0 to disable)')
+    parser.add_argument('--batch-frames-inout', default=0, type=int,
+                        help='Maximum input+output frames in a minibatch (0 to disable)')
+    parser.add_argument('--maxlen-in', default=800, type=int, metavar='ML',
```

Comment:
Can you change these options similar to the ASR ones?

Reply:
ok